services/src/main/java/com/linkedin/kmf/services/configs/ProduceServiceConfig.java

```diff
@@ -122,7 +129,12 @@
                                     2.0,
                                     atLeast(1.0),
                                     ConfigDef.Importance.LOW,
-                                    CommonServiceConfig.PARTITIONS_TO_BROKER_RATIO_DOC);
+                                    CommonServiceConfig.PARTITIONS_TO_BROKER_RATIO_DOC)
+                            .define(TOPIC_FACTORY_CONFIG,
```

Comment:
Type CLASS seems like the correct type to use since it's actually a class name.  The user of the configuration needs to know how to convert the easy to use constants into the class names and it needs to be documented.  The configuration class we use has a nice method that just gives the caller an instance of the specified object.   So it's one line of code vs: 1 line to check its s special constant an "else" if it's not.  When it is a special constant then this needs to be converted into the class name (or just  instantiate directly); 3 lines of code.  I say users can suffer with specifying a fully qualified class name.